echopype/tests/utils/test_processinglevels_integration.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+
+import numpy as np
+import xarray as xr
+import echopype as ep
+
+
+@pytest.mark.parametrize(
+    ["sonar_model", "path_model", "raw_and_xml_paths", "extras"],
+    [
+        (
+            "EK60",
+            "EK60",
+            ("Winter2017-D20170115-T150122.raw", None),
+            {},
+        ),
+        (
+            "AZFP",
+            "AZFP",
+            ("17082117.01A", "17041823.XML"),
+            {"longitude": -60.0, "latitude": 45.0, "salinity": 27.9, "pressure": 59},
+        ),
+    ],
+)
+def test_raw_to_mvbs(
+        sonar_model,
+        path_model,
+        raw_and_xml_paths,
+        extras,
+        test_path
+):
+    # Prepare the Sv dataset
+    raw_path = test_path[path_model] / raw_and_xml_paths[0]
+    if raw_and_xml_paths[1]:
+        xml_path = test_path[path_model] / raw_and_xml_paths[1]
+    else:
+        xml_path = None
+
+    def _presence_test(test_ds, processing_level):
+        assert "processing_level" in test_ds.attrs
+        assert "processing_level_url" in test_ds.attrs
+        assert test_ds.attrs["processing_level"] == processing_level
+
+    def _absence_test(test_ds):
+        assert "processing_level" not in test_ds.attrs
+        assert "processing_level_url" not in test_ds.attrs
+
+    # ---- Convert raw file and update_platform
+    ed = ep.open_raw(raw_path, xml_path=xml_path, sonar_model=sonar_model)
+    if "longitude" in ed['Platform'].data_vars and "latitude" in ed['Platform'].data_vars:
+        _presence_test(ed["Top-level"], "Level 1A")
+    elif "longitude" in extras and "latitude" in extras:
+        _absence_test(ed["Top-level"])
+        point_ds = xr.Dataset(
+            {
+                "latitude": (["time"], np.array([float(extras["latitude"])])),
+                "longitude": (["time"], np.array([float(extras["longitude"])])),
+            },
+            coords={
+                "time": (["time"], np.array([ed["Sonar/Beam_group1"]["ping_time"].values.min()]))
+            },
+        )
+        ed.update_platform(point_ds)
+        _presence_test(ed["Top-level"], "Level 1A")
+    else:
+        _absence_test(ed["Top-level"])
+        raise RuntimeError(
+            "Platform latitude and longitude are not present and cannot be added "
+            "using update_platform based on test raw file and included parameters."
+        )
+
+    # ---- Calibrate and add_latlon
+    env_params = None
+    if sonar_model == "AZFP":
+        # AZFP data require external salinity and pressure
+        env_params = {
+            "temperature": ed["Environment"]["temperature"].values.mean(),
+            "salinity": extras["salinity"],
+            "pressure": extras["pressure"],
+        }
+
+    ds = ep.calibrate.compute_Sv(echodata=ed, env_params=env_params)
+    _absence_test(ds)
+
+    sv_ds = ep.consolidate.add_location(ds=ds, echodata=ed)
+    assert "longitude" in sv_ds.data_vars and "latitude" in sv_ds.data_vars
```

Comment:
I suggest changing `sv_ds` to either `Sv_ds` or `ds_Sv` because Sv and sv unfortunately mean different things in the common notations...

Reply:
Done